src/hypervisor.tex

```diff
@@ -322,15 +322,20 @@ \subsection{Hypervisor Status Register ({\tt hstatus})}
 
 Field GVA (Guest Virtual Address) is written by the implementation
 whenever a trap is taken into HS-mode.
-For any trap (access fault, page fault, or guest-page fault) that writes
+For any trap (breakpoint, address misaligned,
+access fault, page fault, or guest-page fault) that writes
 a guest virtual address to {\tt stval}, GVA is set to~1.
 For any other trap into HS-mode, GVA is set to~0.
 
 \begin{commentary}
-For memory faults, GVA is redundant with field SPV (the two bits are set
+For memory access traps, GVA is redundant with field SPV (the two bits are set
 the same) except when the explicit memory access of an HLV, HLVX, or HSV
 instruction causes a fault.
 In that case, SPV=0 but GVA=1.
+
+If an instruction address misaligned or breakpoint trap writes zero to
+{\tt stval} instead of the faulting virtual address, then GVA=0 even if
+SPV=1.
```

Comment:
Why does this comment refer only to 2 of the 5 types of traps listed just above? (And does this refer only to instruction breakpoint traps and not data breakpoint traps?)

Does this mean that if an access fault, page fault, or guest-page fault that writes 0 to `stval` then GVA can still be 1?

Reply:
For the other types of traps (load/store address misaligned, access fault, page fault, or guest-page fault), the hypervisor extension will require that the appropriate `*tval` CSR must be written with the faulting virtual address.  (This used to be required of all RISC-V cores, but at some point was weakened.)  Because there will be no option to write zero to `*tval` for those traps, GVA will never be zero if SPV = 1.

Reply:
Gotcha...what about data breakpoint vs instruction breakpoint? If this refers to both, how about reversing the two:

> If a breakpoint or instruction address misaligned trap writes zero to {\tt stval} instead of the faulting virtual address, then GVA=0 even if SPV=1.

Reply:
Okay, done.